Sources/FoundationEssentials/String/String+Path.swift

```diff
@@ -22,8 +22,24 @@ import WinSDK
 
 internal import _FoundationCShims
 
+extension StringProtocol {
+    fileprivate func _standardizingSlashes() -> String {
+        #if os(Windows)
+        // The string functions below all assume that the path separator is a forward slash
+        // Standardize the path to use forward slashes before processing for consistency
+        return self.replacing(._backslash, with: ._slash)
+        #else
+        return String(self)
```

Comment:
(nit) is it worth checking whether self is String and just directly return, since most of the extensions below are actually on String?

Reply:
Sure - I updated it to use a specializing cast to determine if `self` is already `String`. I confirmed that when compiling with `-O` the optimizer is smart enough to elide the `_standardizingSlashes()` call entirely when called from an extension on `String`